options_test.go

```diff
@@ -141,3 +141,87 @@ func TestPassAfterNonOptionWithPositionalIntFail(t *testing.T) {
 		assertStringArray(t, ret, test.ret)
 	}
 }
+
+func TestTerminatedOptions(t *testing.T) {
+	type testOpt struct {
+		Args    [][]string `short:"a" long:"args" terminator:";"`
+		Verbose bool       `short:"v"`
+		Mode    int        `short:"m"`
+	}
+
+	tests := []struct {
+		parserOpts Options
+		args       []string
+		_Args      [][]string
+		_Verbose   bool
+		_Mode      int
+		ret        []string
+		shouldErr  bool
+	}{
+		{
+			args: []string{
+				"--args", "bin", "-xyz", "--foo=bar", "-m", "5", "-v", "foo bar", ";",
+				"-v",
+				"--args=", "--no-delim",
+			},
+			_Args: [][]string{
+				{"bin", "-xyz", "--foo=bar", "-m", "5", "-v", "foo bar"},
+				{"--no-delim"},
+			},
+			_Verbose: true,
+		},
+		{
+			args:  []string{"--args", "--foo", "bar;", "-v"},
+			_Args: [][]string{{"--foo", "bar;", "-v"}},
+		},
+		{
+			args:  []string{"--args", "foo\tbar", "\"x y z\""},
+			_Args: [][]string{{"foo\tbar", "\"x y z\""}},
+		},
+		{
+			parserOpts: PassDoubleDash,
+			args:       []string{"--args", "--foo", "--", "bar", ";", "-m", "1", "--", "-v"},
+			_Args:      [][]string{{"--foo", "--", "bar"}},
+			_Mode:      1,
+			ret:        []string{"-v"},
+		},
+		{
+			args:     []string{"-va", "-m", "5"},
+			_Args:    [][]string{{"-m", "5"}},
+			_Verbose: true,
+		},
+	}
+
+	for _, test := range tests {
+		opts := testOpt{}
+		p := NewParser(&opts, test.parserOpts)
+		ret, err := p.ParseArgs(test.args)
+
+		if err != nil {
+			if !test.shouldErr {
```

Comment:
I notice we're never setting `shouldErr` in the test cases. Is there one that can fail? If so, can we add such a case?

Reply:
Added.